lib/minidoc/read_only.rb

```diff
@@ -5,7 +5,15 @@ class Minidoc::ReadOnly < Minidoc::Value
   include Minidoc::Connection
   include Minidoc::Finders
 
-  def self.database
-    Minidoc.database
+  def self.connection
+    super
+  rescue MissingConfiguration, NoMethodError
```

Comment:
This is messier than I would like. The definition of [`class_attribute`](https://github.com/rails/rails/blob/05eb2c08357664ba18c33378fd6110989ed7edca/activesupport/lib/active_support/core_ext/class/attribute.rb#L87-L106) is pretty wacky, and so the method won't actually exist if it hasn't been set already.

And when it does exist, it validates config, so we catch both of these.

The alternative is to check `instance_var_get` directly. I don't know which I like less.

Reply:
Which is the test case where `super` doesn't raise an error? Tried futzing with it a little locally and it seems like it always hits the error

Reply:
`it "can have a separate mongo connection" do`

Reply:
Hmm. Is it possible that we could change the `included` hook that calls `class_attribute`, and define the method in a more-trivial way?

Are we using any of the extra behavior of class_attribute?

Reply:
To show what I mean, this should fail a test, right? (pushed up to `mj/demo` branch)? https://github.com/codeclimate/minidoc/commit/6dc7449c29a55d1a54690a82295e2e67785899e9

Reply:
I can't account for why that `else` isn't working, @maxjacobson: but that test does pass *without* the `rescues`, so it certainly seems to be calling super. Do you want to pair on it?

@wilson that's a good idea, if we can. What other "extra behavior" of `class_attribute` are you referring to?

Reply:
The `else` does not work because `class_attribute`'s `foo=` method is defined to *delete and redefine` any existing reader method for that attribute. So once `Class.collection = ...` is called, the method definition of `self.collection` that includes your `else` is gone forever.

Reply:
Wow.

Reply:
Hehe thanks for explaining that. I like this updated implementation